src/core_modules/capture-core/components/WidgetProfile/getSubValueForTei.js

```diff
@@ -0,0 +1,20 @@
+// @flow
+import { getApi } from 'capture-core/d2';
+import { dataElementTypes } from '../../metaData';
+
+const getImageOrFileResourceSubvalue = async (key: string) => {
+    const { displayName: name } = await getApi().get(`fileResources/${key}`);
+    return name;
+};
+
+
+const getOrganisationUnitSubvalue = async (key: string) => {
+    const { organisationUnits = [] } = await getApi().get(`organisationUnits?filter=id:in:[${key}]`);
```

Comment:
Hey, 
The `setListAttributes` and `getOrganisationUnitSubvalue` are used inside a React component. I think that in this case the DHIS2 hook [useDataQuery](https://runtime.dhis2.nu/#/hooks/useDataQuery) can be used instead of deprecated `getApi`. Do you see any reason why the hooks cannot be used? 
Thanks!

Reply:
hey, I prefer to separate this sub value logic in different file to make it easier to control the element type. `getOrganisationUnitSubvalue` is called conditionally which is not suitable for using hook (even when using `lazy: true`